client/my-sites/checkout/checkout-thank-you/index.tsx

```diff
@@ -275,6 +280,27 @@ export class CheckoutThankYou extends Component<
 		}
 	}
 
+	/**
+	 * Determines whether the current checkout flow is for a redesign V2 purchase.
+	 * Used for gradually rolling out the redesign.
+	 *
+	 * @returns {boolean} True if the checkout flow is for a redesign V2 purchase, false otherwise.
+	 */
+	isRedesignV2 = () => {
+		// Fallback to old design when there is a failed purchase.
+		const failedPurchases = getFailedPurchases( this.props );
+		if ( failedPurchases.length > 0 ) {
+			return false;
+		}
+
+		// ThankYou page for only purchasing a plan.
+		const purchases = getPurchases( this.props );
+		if ( purchases.length === 1 ) {
+			return isWpComPlan( purchases[ 0 ].productSlug );
```

Comment:
We cannot use `isPlan` because it includes Jetpack plans, which we are not supporting. They have a different setup flow.